src/main/java/org/ecocean/servlet/GetCurrentIAInfo.java

```diff
@@ -110,22 +110,44 @@ private JSONObject getIAJSONForAnnotation(Shepherd myShepherd, Annotation ann) {
             annIA.put("assetDetectionStatus", ma.getDetectionStatus());
             annIA.put("assetWebURL", Util.scrubURL(ma.webURL()));
             
-             List<Task> rootTasks = ann.getRootIATasks(myShepherd);
-            if (rootTasks!=null&&!rootTasks.isEmpty()) {
+            //List<Task> rootTasks = ann.getRootIATasks(myShepherd);
+            //if (rootTasks!=null&&!rootTasks.isEmpty()) {
               
               //WB-1812 - previously (commented out right above) we were only ever exposing the first root task
               //instead, let's be consistent with encounterMediaGallery.jsp and return the most recent IA task
-              for (Task t : ma.getRootIATasks(myShepherd)) {
+              /*for (Task t : ma.getRootIATasks(myShepherd)) {
                   if (rootTasks.contains(t)) continue;
                   if (t.deepContains(ann)!=null) rootTasks.add(t);
               }
-              Comparator<Task> byRanking = 
-                  (Task tsk1, Task tsk2) -> Long.compare(tsk1.getCreatedLong(), tsk2.getCreatedLong());
-              Collections.sort(rootTasks, byRanking);
-              Collections.reverse(rootTasks); // now desc, ez
-              annIA.put("lastTaskId", rootTasks.get(0).getId());
+              */
+               //let's look for match results we can easily link for the user
+               ArrayList<Task> tasks=new ArrayList<Task>();
+               List<Task> relatedTasks = Task.getTasksFor(ann, myShepherd);
+               if(relatedTasks!=null && relatedTasks.size()>0){
+                 for(Task task:relatedTasks){  
+                   if(task.getParent()!=null && task.getParent().getChildren().size()==1){
```

Comment:
We could do something like
Boolean isTaskWithOnlyOneAlgorithm = task.getParent()!=null && task.getParent().getChildren().size()==1

And then do if(isTaskWithOnlyOneAlgorithm)... Much more self-documenty.

Reply:
It would be clear, but it would not be reused, and the next bit of logic still has conditions requiring checks that are not the opposite of this boolean.